File: src/main/java/me/alegian/thavma/impl/common/menu/slot/T7Slot.java
package me.alegian.thavma.impl.common.menu.slot;

import me.alegian.thavma.impl.common.menu.Menu;
import net.minecraft.world.Container;
import net.minecraft.world.inventory.Slot;
import org.jetbrains.annotations.NotNull;

/**
 * A Slot that implements Sized, and has a back-reference to a Menu
 */
public class T7Slot<T extends Menu> extends Slot implements DynamicSlot<T> {
  private int size = 0;
  private final T menu;
  private float x,y;

  public T7Slot(Container container, int id, T menu) {
    super(container, id, 0, 0);
    this.menu = menu;
  }

  @Override
  public @NotNull T getMenu() {
    return this.menu;
  }

  @Override
  public float getActualX() {
    return x;
  }

  @Override
  public float getActualY() {
    return y;
  }

  @Override
  public void setActualX(float x) {
    this.x = x;
  }

  @Override
  public void setActualY(float y) {
    this.y = y;
  }

  @Override
  public int getSize() {
    return this.size;
  }

  @Override
  public void setSize(int size) {
    this.size = size;
  }
}

File: src/main/java/me/alegian/thavma/impl/common/menu/slot/T7Slot.kt
package me.alegian.thavma.impl.common.menu.slot

import me.alegian.thavma.impl.common.menu.Menu
import net.minecraft.world.Container
import net.minecraft.world.inventory.Slot

/**
 * A Slot that implements Sized, and has a back-reference to a Menu
 */
open class T7Slot<T : Menu>(container: Container, id: Int, override val menu: T) : Slot(container, id, 0, 0), DynamicSlot<T> {
  override var size = 0
  override var actualX = 0f
  override var actualY = 0f
}
